### PR TITLE
adds basic deploy template for frontend

### PR DESCRIPTION
**Pull request by tonytheleg**
### What this PR does
**Before this PR:**

* no manifests/process for deploying frontend into a cluster

**After this PR:**

* adds a basic k8s template to deploy frontend as a deployment
* adds make targets to easily deploy/undeploy frontend in a cluster, leveraging existing parameters/defaults
* updates make variable `ARO_HCP_BASE_IMAGE` to be configurable for testing since there is no ACR yet
* updates frontend README with new make targets for visibility

Jira: 
Link to demo recording: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Deployment: The deployment process was considered and addressed if required
- [ ] Testing: New code requires new unit tests.
- [ ] Documentation: Is the documentation updated? Either in the doc located in focus area, in the README or in the code itself.
- [ ] Customers: Is this change affecting customers? Is the release plan considered?

### CODE REVIEW

**Comment by tonytheleg** [comment]
@microsoft-github-policy-service agree [company="Red Hat"]

**Comment by tonytheleg** [comment]
@microsoft-github-policy-service agree company="Red Hat"

File: Makefile [hunk 1]
```diff
@@ -34,3 +34,9 @@ frontend-multistage:
 
 clean:
 	rm aro-hcp-frontend
+
+deploy-frontend:
+	oc process -f ./deploy/aro-hcp-frontend.yml -p ARO_HCP_FRONTEND_IMAGE=${ARO_HCP_FRONTEND_IMAGE} | oc create -f -
```
Comment (mjlshen) [inline Makefile:39]:
```suggestion
	oc process -f ./deploy/aro-hcp-frontend.yml -p ARO_HCP_FRONTEND_IMAGE=${ARO_HCP_FRONTEND_IMAGE} | oc replace -f -
```
Nit, suggesting `replace`/`apply` to make it idempotent

**Review by mjlshen — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (tonytheleg) [reply Makefile]: Updated

**Review by mjlshen — APPROVED** [review]
/lgtm

**Pull request merged**